pkg/server/share_manager.go

```diff
@@ -210,6 +240,96 @@ func (m *ShareManager) resizeVolume(devicePath, mountPath string) error {
 	return nil
 }
 
+func (m *ShareManager) makeLeaseClient() error {
+	cfg, err := rest.InClusterConfig()
+	if err != nil {
+		m.logger.WithError(err).Warn("Failed to create InClusterConfig")
+		return err
+	}
+
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		m.logger.WithError(err).Warn("Failed to create kubernetes client set")
+		return err
+	}
+
+	// Use the clientset to get the node name of the share-manager pod
+	// and store for use by takeLease().
+	podName := shareManagerPrefix + m.volume.Name
+	pod, err := clientset.CoreV1().Pods(namespace).Get(m.context, podName, metav1.GetOptions{})
+	if err == nil && pod != nil {
+		m.leaseHolder = pod.Spec.NodeName
+	} else {
+		m.leaseHolder = podName
+	}
+
+	m.leaseClient = clientset.CoordinationV1()
+	return nil
+}
+
+func (m *ShareManager) getLease() error {
+	if m.leaseClient == nil {
+		return fmt.Errorf("kubernetes API client is unset")
+	}
+
+	lease, err := m.leaseClient.Leases(namespace).Get(m.context, m.volume.Name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	m.lease = lease
+	return nil
+}
+
+func (m *ShareManager) takeLease() error {
+	now := time.Now()
+
+	currentHolder := *m.lease.Spec.HolderIdentity
+	if currentHolder != "" {
+		m.logger.Warnf("Lease holderIdentity was not cleared - replacing %v with %v", currentHolder, m.leaseHolder)
```

Comment:
It is normal that lease's holder identity is not updated at this point. I am thinking to change the sentiment of the log message a bit
```suggestion
		m.logger.Warnf("Updating lease holderIdentity from %v to %v", currentHolder, m.leaseHolder)
```